Development/src/components/ActiveField.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect } from 'react';
+import { Switch } from '@material-ui/core';
+import { useNotify } from 'react-admin';
+import get from 'lodash/get';
+import omit from 'lodash/omit';
+import dataProvider from '../dataProvider';
+
+const sanitizeRestProps = props =>
+    omit(props, [
+        'addLabel',
+        'allowEmpty',
+        'basePath',
+        'cellClassName',
+        'className',
+        'formClassName',
+        'headerClassName',
+        'label',
+        'linkType',
+        'link',
+        'locale',
+        'record',
+        'resource',
+        'sortable',
+        'sortBy',
+        'source',
+        'textAlign',
+        'translateChoice',
+    ]);
+
+const toggleMasterEnable = (record, resource) => {
+    return new Promise((resolve, reject) =>
+        dataProvider('GET_ONE', resource, {
+            id: record.id,
+        })
+            .then(({ data }) => {
+                const params = {
+                    id: get(data, 'id'),
+                    data: {
+                        ...data,
+                        $staged: {
+                            ...get(data, '$staged'),
+                            master_enable: !get(data, '$staged.master_enable'),
+                            activation: { mode: 'activate_immediate' },
+                        },
+                    },
+                    previousData: data,
+                };
+                return dataProvider('UPDATE', resource, params);
+            })
+            .then(() => resolve())
+            .catch(error => reject(error))
+    );
+};
+
+const ActiveField = ({ className, source, record = {}, resource, ...rest }) => {
+    const notify = useNotify();
+    const [checked, setChecked] = React.useState(
+        get(record, 'subscription.active')
+    );
+
+    const handleChange = (record, resource) => {
+        toggleMasterEnable(record, resource)
+            .then(() => setChecked(!checked))
+            .catch(error => notify(error.toString(), 'warning'));
+    };
+
+    useEffect(() => {
```

Comment:
Would it be worth a comment here to say why we're doing this rather than refresh?

Reply:
That comment doesn't make sense to me, but that may be unfamiliarity with React terminology... what do 'refresh' and 'sent' mean in this instance?

Reply:
Let me reword this, then I'll force push over this commit with the other changes.
Does this make more sense?
When the page refresh button is pressed, the ActiveField will receive a new record prop. When this happens we should update the state of the switch to reflect the newest IS-04 data

Reply:
Oh, hmm, so this is about page refresh? I kinda assumed that the individual components got torn down at that point, and the table page would be rebuilt from a new response, so I didn't see why there would be special handling for the `ActiveField`. I guess I was wrong, and because we've allowed the component's state to be tweaked, we need to 'clear' that?

Reply:
To prevent a jerky page movement I'm updating all components in place yes. If I wanted to fully reset the components I could tear them down as you were saying. This is functionally equivalent as we reset the component state to it's initial value.
The specific case I was targeting was if an external factor has altered the IS-04 data such as another client.

Reply:
And by 'initial value' you mean "new value, but without being masked by the 'local' change made by toggling the switch"?

Reply:
Yes, that's a better way of putting it

Reply:
Perfect